module-apps/apps-common/widgets/spinners/Container.hpp

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#pragma once
+
+#include <vector>
+#include <string>
+#include <type_traits>
+
+/// Generic elements container.
```

Comment:
just nitpicking but isn't it more like a model than a container? :) It wraps the generic stl container and provides user-specific stored data to the view, doesn't it?

In the listview, we call such an item provider. Would you consider taking a quick peek there too? just for reference.

Reply:
Well, actually it's not limited to the GUI. It can contain anything you want. Its main purpose is to provide specific iteration model (fixed/rotating).

Reply:
it's in widgets and used only for widgets built on top of actually generic (stl) containers using integers instead iterators. If you wish for it's generic use - then use to utils as it's in the wrong place then?
Although in my opinion it would be a bit faking what it's not :) Saying that - I do not care much. Do whatever you want here.

Reply:
Fixed :)